src/Data/Fin/Properties.agda

```diff
@@ -68,6 +68,9 @@ private
 nonZeroIndex : Fin n → ℕ.NonZero n
 nonZeroIndex {n = suc _} _ = _
 
+zeroFromNonZero : .⦃ _ : ℕ.NonZero n ⦄ → Fin n
+zeroFromNonZero {n = suc _} = zero
```

Comment:
Anyone have any bright ideas for what this could be called? @jamesmckinna etc?

It should also live in `Data.Fin.Base` I think?

Reply:
So, the abandoned #1847 , tackling #1686 , called this concept "homogeneous zero" (and its friend "homogeneous successor", on the model of `n : Nat, i : Fin n` being a 'homogeneous' telescope; of course, the `{{NonZero n}}` precondition spoils things, but hey ;-)), and gave it the terrible name `zero\'` in the PR. 

:+1: to its being in `Data.Fin.Base`, as in that PR.

Better name(s): `hzero`? `zeroNZ`? `zero⁺`? 

I ran out of mental strength/inspiration working on that PR, and I find it slightly hard to revisit the discussion/thought processes again now...